Lib/test/test_subprocess.py

```diff
@@ -786,6 +786,19 @@ def test_env(self):
             stdout, stderr = p.communicate()
             self.assertEqual(stdout, b"orange")
 
+    @unittest.skipUnless(sys.platform == "win32", "Windows only issue")
+    def test_win32_duplicate_envs(self):
+        newenv = os.environ.copy()
+        newenv["fRUit"] = "cherry"
+        newenv["fruit"] = "lemon"
+        newenv["FRUIT"] = "orange"
+        newenv["frUit"] = "banana"
+        with subprocess.Popen(["CMD", "/c", "SET", "fruit"],
+                              stdout=subprocess.PIPE,
+                              env=newenv) as p:
+            stdout, stderr = p.communicate()
+            self.assertEqual(stdout, b"fRUit=cherry\r\n")
```

Comment:
```suggestion
            self.assertEqual(stdout.strip(), b"frUit=banana")
```

I think we want the last assigned value. That may not be how it's working today, but there's no way people won't be surprised by not getting the value they set (and a much more likely scenario is `{**os.environ, "Path": path}` not overriding the system's `"Path"` which was already normalised to `"PATH"` and causing the wrong apps to run).

Also, `.strip()` rather than comparing the newline avoids encoding knowledge about exactly what output will also be returned (e.g. we might get two newlines on some systems, or in future changes, etc.)